app/src/main/java/org/thoughtcrime/securesms/contactshare/ContactModelMapper.java

```diff
@@ -0,0 +1,172 @@
+      for (SharedContact.Email email : sharedContact.getEmail().get()) {
+        emails.add(new Email(email.getValue(),
+                             remoteToLocalType(email.getType()),
+                             email.getLabel().orNull()));
+      }
+    }
+
+    List<PostalAddress> postalAddresses = new LinkedList<>();
+    if (sharedContact.getAddress().isPresent()) {
+      for (SharedContact.PostalAddress postalAddress : sharedContact.getAddress().get()) {
+        postalAddresses.add(new PostalAddress(remoteToLocalType(postalAddress.getType()),
+                                              postalAddress.getLabel().orNull(),
+                                              postalAddress.getStreet().orNull(),
+                                              postalAddress.getPobox().orNull(),
+                                              postalAddress.getNeighborhood().orNull(),
+                                              postalAddress.getCity().orNull(),
+                                              postalAddress.getRegion().orNull(),
+                                              postalAddress.getPostcode().orNull(),
+                                              postalAddress.getCountry().orNull()));
+      }
+    }
+
+    Avatar avatar = null;
+    if (sharedContact.getAvatar().isPresent()) {
+      Attachment attachment = PointerAttachment.forPointer(Optional.of(sharedContact.getAvatar().get().getAttachment().asPointer())).get();
+      boolean    isProfile  = sharedContact.getAvatar().get().isProfile();
+
+      avatar = new Avatar(null, attachment, isProfile);
+    }
+
+    return new Contact(name, sharedContact.getOrganization().orNull(), phoneNumbers, emails, postalAddresses, avatar);
+  }
+
+  private static Phone.Type remoteToLocalType(SharedContact.Phone.Type type) {
+    switch (type) {
+      case HOME:   return Phone.Type.HOME;
+      case MOBILE: return Phone.Type.MOBILE;
+      case WORK:   return Phone.Type.WORK;
+      default:     return Phone.Type.CUSTOM;
+    }
+  }
+
+  private static Email.Type remoteToLocalType(SharedContact.Email.Type type) {
+    switch (type) {
+      case HOME:   return Email.Type.HOME;
+      case MOBILE: return Email.Type.MOBILE;
+      case WORK:   return Email.Type.WORK;
+      default:     return Email.Type.CUSTOM;
+    }
+  }
+
+  private static PostalAddress.Type remoteToLocalType(SharedContact.PostalAddress.Type type) {
+    switch (type) {
+      case HOME:   return PostalAddress.Type.HOME;
+      case WORK:   return PostalAddress.Type.WORK;
+      default:     return PostalAddress.Type.CUSTOM;
+    }
+  }
+
+  private static SharedContact.Phone.Type localToRemoteType(Phone.Type type) {
+    switch (type) {
+      case HOME:   return SharedContact.Phone.Type.HOME;
+      case MOBILE: return SharedContact.Phone.Type.MOBILE;
+      case WORK:   return SharedContact.Phone.Type.WORK;
+      default:     return SharedContact.Phone.Type.CUSTOM;
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 148:153, 157:162.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.